PLFM-9088

nickgros opened the pull request:
- Update PatchRowHandler to include metadata with each row. Currently, the metadata includes the information needed to map the grid row back to a row in a Synapse table or view.
- Fix minor documentation issue in Row.json
- Unconditionally retrieve etags in queries used to build a grid session

nickgros committed:
PLFM-9088

- Update PatchRowHandler to include metadata with each row. Currently, the metadata includes the information needed to map the grid row back to a row in a Synapse table or view.
- Fix minor documentation issue in Row.json
- Unconditionally retrieve etags in queries used to build a grid session

nickgros committed:
fix integration test

john-hill commented on services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/grid/PatchRowHandler.java:
```diff
@@ -99,20 +99,130 @@ public PatchRowHandler(PatchStore patchStore, String sessionId, Long replicaId,
+		Long rowId = row.getRowId();
+		if (rowId != null) {
+			NewConstant rowIdConst = currentPatch.addNewOperation(NewConstant.class)
+					.setValue(new ConValue(ConType.DOUBLE, rowId));
```
Long.

john-hill commented on services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/grid/PatchRowHandler.java:
```diff
@@ -99,20 +99,130 @@ public PatchRowHandler(PatchStore patchStore, String sessionId, Long replicaId,
+
+		}
+
+		String etag = row.getEtag();
```
ensure this works for Table.

john-hill commented on services/repository-managers/src/test/java/org/sagebionetworks/repo/manager/grid/PatchRowHandlerTest.java:
```diff
@@ -53,7 +53,7 @@ public void testNoColumnsNoRows() throws IOException {
 		verify(mockStore, times(1)).savePatch(any(), any(), any());
 		// This patch has been tested with the JSON-Joy TypeScript library.
 		verify(mockStore).savePatch(sessionId, new LogicalTimestamp().setReplicaId(replicaId).setSequenceNumber(1L),
-				"[[[19,1]],[2],[0,\"0.0.2\"],[3],[6],[6],[10,1,[[\"doc_version\",2],[\"columnNames\",3],"
```
0.1.0 since this is a breaking change.

nickgros committed:
code review changes

- Use LONG instead of DOUBLE for integer constants
- Change model version to 0.1.0
